### PR TITLE
Update Credentials.php

### PR DESCRIPTION
**Pull request by alirezarahmani**
I think you forget make them private

### CODE REVIEW

**Comment by jorge07** [comment]
Hi @alirezarahmani. Thanks for open this pull request. This is an interesting topic. This PR make sense, it will avoid mutations in the object so we can avoid side effects.
But in another hand, when an object is used only as dto (data transfer object), like this case or the application commands, is common make the properties public so you don't need to define the getter. It avoid lot of "useless" code in large codebases but it need the commitment of all the team members to never, never mutate an attribute.

In open source projects, it totally make sense, as in php the attributes can't be defined as `readonly` like in typescript, so it will avoid side effects.

This PR need return types in the getters, probably some more refactor in the places this object is used and fix the tests.

File: src/Domain/User/ValueObject/Auth/Credentials.php [hunk 1]
```diff
@@ -11,16 +11,26 @@ class Credentials
...
-    public $password;
+    private $password;
 
     public function __construct(Email $email, HashedPassword $password)
     {
         $this->email = $email;
         $this->password = $password;
     }
+    
+    public function email()
```
Comment (jorge07) [inline src/Domain/User/ValueObject/Auth/Credentials.php]: needs return types

File: src/Domain/User/ValueObject/Auth/Credentials.php [hunk 2]
```diff
@@ -11,16 +11,26 @@ class Credentials
...
+    {
+        return $this->email;
+    }
+    
+    public function password()
```
Comment (jorge07) [inline src/Domain/User/ValueObject/Auth/Credentials.php]: needs return types

**Comment by cermakondrej** [comment]
@jorge07 Also I would like to point out that yout current code is only working thanks to php not being strict on typing.

When you execute `GetTokenHandler`, your `MysqlUserReadModelRepository::getCredentialsByEmail` executes `oneByEmail`, which returns `UserView`. However, the `Credentials` ValueObject inside hasn't been properly hydrated and which result into something like this:
Screenshot from 2020-02-18 22-42-09

Which means, that if those properties would be accessed with getters, this particular UseCase Would break down due to this hydration problem :)

**Pull request closed without merging**